### PR TITLE
Add MQTT generate token implementation

### PR DESCRIPTION
**Pull request by cqnguy23**
# Description

1. Support to generate MQTT JWT Token for Azure WebPubSub. Essentially we need to identify if clients want to establish a default WebPubSub connection or an MQTT connection, then we will generate the Client Connection URL for them accordingly.

- Example Client URL for:
  - Default Connection: `wss://exampleHost.com/client/hubs/exampleHub`
  - MQTT Connection: `wss://exampleHost.com/clients/mqtt/hubs/exampleHub`
 
2. Upgrade API version to '2024-01-01'
3. Added operations `add_connections_to_groups` and `remove_connections_from_group`
4. Add tests to get MQTT Token

### CODE REVIEW

File: sdk/webpubsub/azure-messaging-webpubsubservice/tests/test_reverse_proxy_async.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ def _callback(pipeline_request):
         wps_endpoint = "https://wps.contoso.com/"
         apim_endpoint = "https://apim.contoso.com/"
         credential = AzureKeyCredential("AzureKeyCredential")
-        request = build_send_to_all_request("Hub", content='test_webpubsub_send_request', content_type='text/plain')
+        request = build_web_pub_sub_service_send_to_all_request("Hub", content='test_webpubsub_send_request', content_type='text/plain')
```
Comment (johanste) [inline sdk/webpubsub/azure-messaging-webpubsubservice/tests/test_reverse_proxy_async.py:28]: Why is this test using the send_request method? If that is what we expect customers to use, why are we adding client level method for this?

Comment (cqnguy23) [reply sdk/webpubsub/azure-messaging-webpubsubservice/tests/test_reverse_proxy_async.py:28]: the send_to_all method that the customers use are tested in the other file I believe (test_smoke_py)

File: sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/_operations/_patch.py [hunk 2]
```diff
@@ -80,7 +83,7 @@ def get_token_by_key(endpoint: str, hub: str, key: str, **kwargs: Any) -> str:
 
 class WebPubSubServiceClientOperationsMixin(WebPubSubServiceClientOperationsMixinGenerated):
     @distributed_trace
-    def get_client_access_token(self, **kwargs: Any) -> JSON:
+    def get_client_access_token(self, *, webpubsub_client_access: Optional[str] = "Default", **kwargs: Any) -> JSON:
```
Comment (johanste) [inline sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/_operations/_patch.py:86]: The name of the parameter and the comment are inconsistent. Why not name the parameter `client_type` using the nomenclature in the comment?

File: sdk/webpubsub/azure-messaging-webpubsubservice/tests/test_jwt.py [hunk 3]
```diff
@@ -117,3 +117,22 @@ def test_pass_in_jwt_headers(connection_string):
     kid = '1234567890'
     token = client.get_client_access_token(jwt_headers={"kid":kid })['token']
     assert jwt.get_unverified_header(token)['kid'] == kid
+
+test_cases = [
+    ("Endpoint=http://localhost;Port=8080;AccessKey={};Version=1.0;".format(access_key), "hub", "ws://localhost:8080/clients/mqtt/hubs/hub"),
+    ("Endpoint=https://a;AccessKey={};Version=1.0;".format(access_key), "hub", "wss://a/clients/mqtt/hubs/hub"),
+    ("Endpoint=http://a;AccessKey={};Version=1.0;".format(access_key), "hub", "ws://a/clients/mqtt/hubs/hub")
+]
+@pytest.mark.parametrize("connection_string,hub,expected_url", test_cases)
```
Comment (johanste) [inline sdk/webpubsub/azure-messaging-webpubsubservice/tests/test_jwt.py:126]: I would expect more test cases for something that does url manipulation and parsing....

File: sdk/webpubsub/azure-messaging-webpubsubservice/tests/test_reverse_proxy_async.py (hunk 1 above)
Comment (johanste) [reply sdk/webpubsub/azure-messaging-webpubsubservice/tests/test_reverse_proxy_async.py:28]: Ok - I generally prefer testing through the public API surface area.

File: sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/_operations/_patch.py (hunk 2 above)
Comment (cqnguy23) [reply sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/_operations/_patch.py:86]: So we had a discussion with the C# SDK owner and decided that client_type might be a bit too generic. So we decided to use webpubsub_client_protocol (which I'm going to change now)

File: sdk/webpubsub/azure-messaging-webpubsubservice/tests/test_jwt.py (hunk 3 above)
Comment (cqnguy23) [reply sdk/webpubsub/azure-messaging-webpubsubservice/tests/test_jwt.py:126]: Sure @johanste. There are actually already other test cases for this method `get_client_access_token` within this file `test_jwt_py` already. I'm just adding more cases relating to the new MQTT token parser logic

File: sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/_operations/_patch.py (hunk 2 above)
Comment (cqnguy23) [reply sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/_operations/_patch.py:86]: So internally we can leave it as client_type, but for consistency across the SDKs, we're gonna leave it as `webpubsub_client_protocol `

**Review by johanste — APPROVED** [review]

File: sdk/webpubsub/azure-messaging-webpubsubservice/CHANGELOG.md [hunk 4]
```diff
@@ -1,14 +1,11 @@
 # Release History
 
-## 1.1.1 (Unreleased)
+## 1.2.0 (2024-07-03)
```
Comment (msyyc) [inline sdk/webpubsub/azure-messaging-webpubsubservice/CHANGELOG.md:3]: Overall looks good and please update date before merge.

**Review by msyyc — APPROVED** [review]

**Pull request merged**